src/libs/OptionsListUtils.js

```diff
@@ -231,7 +231,8 @@ function createOption(logins, personalDetails, report, {
     let text;
     let alternateText;
     if (isChatRoom || isPolicyExpenseChat) {
-        text = (isArchivedRoom && report.isOwnPolicyExpenseChat) ? report.oldPolicyName : lodashGet(report, ['reportName'], '');
+        text = lodashGet(report, 'reportName')
+            || lodashGet(report, 'oldPolicyName', '');
```

Comment:
Good question – this probably needs to be cleaned up. The `reportName` for archived "own PolicyExpenseChat` is just ` (archived)`, which surely will lead to unexpected behavior. I think if we update this to make sure the `reportName` is correct in Onyx, then we could just use that here directly instead of `oldPolicyName`.

Reply:
There's a lot of overlap between [`getChatReportName`](https://github.com/Expensify/App/blob/a1672dd1f513d3a80c83512bbc4146d604e5a7c5/src/libs/actions/Report.js#L146-L179) and [`ReportUtils.getTitle`](https://github.com/Expensify/App/blob/9b23765efa396d7d3e22d8949c6604d77b62afef/src/libs/ReportUtils.js#L440-L478).

I think I'll refactor this to:

1. Rename `ReportUtils.getTitle` to `ReportUtils.getReportName`.
1. Add more unit tests for `ReportUtils.getReportName` to add separate tests for user-created policy rooms and default policy rooms.
1. Refactor things to eliminate `getChatReportName` and use `ReportUtils.getReportName` instead.
1. In the UI, just reference the `reportName` in Onyx directly instead of using `ReportUtils.getReportName`.

Does that sound like a good plan?

Reply:
>make sure the reportName is correct in Onyx, then we could just use that here directly instead of oldPolicyName

That makes sense to me. But I also curious if we can provide the correct `reportName` via the server. Why do these reports have names that are not the actual name we want to show... 🤔

Reply:
> But I also curious if we can provide the correct reportName via the server. Why do these reports have names that are not the actual name we want to show

So I did a fair amount of research into what it would look like to generate the `reportName` for chat reports via the server, and I don't think it's a good idea. Report titles are generated from a variety of sources / pieces of information:

- The report itself
- Who is looking at the report? (admin view of workspace chat is the employee name, member view is the policy name)
    - Because the title is different depending on who's looking at it, we can't really save the correct report name in the database.
- personal details
- policies

The server _can_ do all the logic to generate the correct title using information from the current user, report, policy, and personalDetails. But if we do that we lose all the benefits of "live" Pusher data.
    - i.e: if report titles are generated only on the server and a report participant updates their personal details, you would only see that change if you refresh the page or fetch the report again.
    - If report titles are generated on the front-end, then you would see that changes as soon as the personal details for that user update (currently we don't have a Pusher channel for personal details, but hopefully we will soon).
    
Does that make sense?

Reply:
Honestly kept going back and forth on this, so I posted in slack [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1650935122490379)

Reply:
Yeah it all makes sense. I guess a counter to the approach is that if the server is supposed to be pushing changes directly to Onyx (a la Optimized API) then maybe a change to personal details could also send out updates about report names. But doesn't seem like there is a ton of clear value in doing that.